zkvm/src/vm.rs

```diff
@@ -537,18 +548,40 @@ where
             Item::Value(value) => Ok(WideValue {
                 r1cs_qty: self.attach_variable(value.qty).1,
                 r1cs_flv: self.attach_variable(value.flv).1,
-                // TBD: add witness for Value types where it exists.
-                witness: None,
+                witness: match (
+                    self.variable_assignment(value.qty),
+                    self.variable_assignment(value.flv),
+                ) {
+                    (Some(ScalarKind::Integer(q)), Some(ScalarKind::Scalar(f))) => Some((q, f)),
+                    (None, None) => None,
+                    (_, _) => return Err(VMError::FormatError),
```

Comment:
AFAIR, there are a few places where assignments may be inconsistent. Would it make sense to introduce VMError::InconsistentWitness?